src/index.js

```diff
@@ -0,0 +1,54 @@
+'use strict'
+
+const debug = require('debug')
+const log = debug('bitswap')
+log.error = debug('bitswap:error')
+
+// const cs = require('./constants')
+// const WantManager = require('./want-manager')
+const decision = require('./decision')
+
+module.exports = class Bitwap {
+  constructor (p, network, bstore) {
+    // the ID of the peer to act on behalf of
+    this.self = p
+
+    // the network delivers messages
+    this.network = network
+
+    // local database
+    this.blockstore = bstore
+
+    // handle message sending
+    // this.wm = new WantManager(network)
+
+    this.engine = new decision.Engine(bstore)
+
+    // this.wm.run()
+  }
+
+  // handle messages received through the network
+  _receiveMessage (p, incoming) {
+
+  }
+
+    // getBlock attempts to retrieve a particular block with key `k` from peers
+  getBlock (k) {
+    throw new Error('Not implemented')
+  }
```

Comment:
note: event emitter that is not an event emitter as discussed with @dignifiedquire